src/main/java/uk/gov/hmcts/dts/fact/services/CourtService.java

```diff
@@ -170,6 +171,17 @@ public ServiceAreaWithCourtReferencesWithDistance getNearestCourtsByPostcodeSear
         return new ServiceAreaWithCourtReferencesWithDistance(serviceArea, convert(courts));
     }
 
+    public ServiceAreaWithCourtReferencesWithDistance getNearestCourtsByAreaOfLawSinglePointOfEntry(final String postcode, final String serviceArea, final String areaOfLaw) {
+        ServiceAreaWithCourtReferencesWithDistance results = this.getNearestCourtsByPostcodeSearch(postcode, serviceArea);
+        results.setCourts(results.getCourts()
```

Comment:
One to consider: is it more performant to create another repo method that searches by postcode, service area, area of law, or is it more performant to stream through a list of courts and check that way? 

Prob more performant to do it on the database, but we will likely only be getting 5 or so results back, so probably isn't a biggy

Reply:
results are always low in numbers so believe should be ok but can revise in future if possible permanence issues arise.